portfolio/src/main/java/com/google/sps/servlets/Comments.java

```diff
@@ -0,0 +1,191 @@
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter an integer larger than -1.");
+      return;
+    }
+    else{
+      // Loop over the entities to store the feedback in a list
+      List<Feedback> statements = new ArrayList<Feedback>();
+      for (Entity entity : results.asIterable()) {
+        if(userChoice == 0) {
+          break;
+        }
+        String name = (String) entity.getProperty("Name");
+        String email = (String) entity.getProperty("Email");
+        String comment = (String) entity.getProperty("Comment");
+        long timestamp = (long) entity.getProperty("Timestamp");
+	    
+        Feedback feedback = new Feedback(name, email, comment, timestamp);
+        statements.add(feedback);
+        userChoice--;
+      }
+      // Send the JSON as the response
+      Gson gson = new Gson();
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(statements));
+    }
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    // Get the body of the HTTP Post
+    String body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+
+    Gson gson = new Gson();
+    Comment target = gson.fromJson(body, Comment.class);
+
+    // Get the input from the form
+    String name = target.name;
+    String email = userService.getCurrentUser().getEmail();
+    String comment = target.comment;
+    long timestamp = System.currentTimeMillis();
+
+    // Create an entity and set its properties
+    Entity dataEntity = new Entity("Data");
+    dataEntity.setProperty("Name", name);
+    dataEntity.setProperty("Email", email);
+    dataEntity.setProperty("Comment", comment);
+    dataEntity.setProperty("Timestamp", timestamp);
+
+    // Store the entities
+    datastore.put(dataEntity);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(dataEntity));
+  }
+  
+  /** 
+   * Returns the number of comments the user wants, or -1 if the choice was invalid.
+   */
+  private int getNumberOfComments(HttpServletRequest request){
+    // Get the input from the form.
+    String userChoiceString = request.getParameter("numComments");
+	
+    // Convert the input to an int.
+    int userChoice;
+    try {
+      userChoice = Integer.parseInt(userChoiceString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + userChoiceString);
+      return Integer.MAX_VALUE;
+    }
+
+    // Check that the input is greater than 0.
```

Comment:
The comment says > 0, but the if statement allows 0 as a valid value. Maybe >= 0?